website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+be able to communicate with Nomad's HTTP API when run on the same host. Other
+SANs may be added including a DNS resolvable hostname to allow remote HTTP
+requests from third party tools.
+
+You should now have the following files:
+
+* `cfssl.json` - cfssl configuration.
+* `nomad-ca.csr` - CA signing request.
+* `nomad-ca-key.pem` - CA private key. Keep safe!
+* `nomad-ca.pem` - CA public certificate.
+* `cli.csr` - Nomad CLI certificate signing request.
+* `cli.pem` - Nomad CLI certificate.
+* `cli-key.pem` - Nomad CLI private key.
+* `client.csr` - Nomad client node certificate signing request for the `global` region.
+* `client-key.pem` - Nomad client node private key for the `global` region.
+* `client.pem` - Nomad client node public certificate for the `global` region.
+* `server.csr` - Nomad server node certificate signing request for the `global` region.
+* `server-key.pem` - Nomad server node private key for the `global` region.
+* `server.pem` - Nomad server node public certificate for the `global` region.
+
+Each Nomad node should have the appropriate key (`-key.pem`) and certificate
+(`.pem`) file for its region and role. In addition each node needs the CA's
+public certificate (`nomad-ca.pem`).
+
+## Configuring Nomad
+
+Once you have the appropriate key and certificates installed you're ready to
+configure Nomad to use them for mTLS. Starting with the [server configuration
+from the Getting Started guide][guide-server] add the following TLS
+CONFIGUration options:
+
+```hcl
+# Increase log verbosity
+log_level = "DEBUG"
+
+# Setup data dir
+data_dir = "/tmp/server1"
+
+# Enable the server
+server {
+    enabled = true
+
+    # Self-elect, should be 3 or 5 for production
+    bootstrap_expect = 1
+}
+
+# Require TLS
+tls {
+  http = true
+  rpc  = true
+
+  ca_file   = "nomad-ca.pem"
+  cert_file = "server.pem"
+  key_file  = "server-key.pem"
+
+  verify_server_hostname = true
+  verify_https_client    = true
+}
+```
+
+The new `tls` section is worth breaking down in more detail:
+
+```hcl
+  http = true
+  rpc  = true
+```
+
+This enables TLS for the HTTP and RPC protocols. Unlike web servers, Nomad
+doesn't use separate ports for TLS and non-TLS traffic: your cluster should
+either use TLS or not.
+
+```hcl
+  ca_file   = "nomad-ca.pem"
+  cert_file = "server.pem"
+  key_file  = "server-key.pem"
+```
+
+The file lines should point to whereever you placed the certificate files on
+the node. This guide assumes they are in Nomad's current directory.
+
+```hcl
+  verify_server_hostname = true
+  verify_https_client    = true
+```
+
+These two settings are important for ensuring all of Nomad's mTLS security
+properties are met. If `verify_server_hostname` is set to `false` the node's
+cerificate will be checked to ensure it is signed by the same CA, but its role
+and region will not be verified. This means any service with a certificate from
+the same CA as Nomad can act as a client or server of any region.
```

Comment:
Changed `from the` to `signed by`:

> This means any service with a certificate signed by same CA as Nomad can act as a client or server of any region.

Not sure how to clarify "service." I mean literally anything listening on a socket with a certificate signed by the same CA poses a threat if this setting is disabled. "service" seemed like the most generic term for that.